### DISCUSSION TITLE
Please remove the VOLUME definition in the official Dockerfile

### DISCUSSION THREAD

**Discussion by AKockWorth** [directus/directus#10402] [Feature Requests]
Due to the issue described here: https://boxboat.com/2017/01/23/volumes-and-dockerfiles-dont-mix/

> If you happen to be unlucky enough to base your image off of another that defines a volume in their Dockerfile, there's no reliable way to update the contents of the volume contents

I cannot create a derived image and add/manipulate files within the `directus/extensions` directory.

**Comment by rijkvanzanten** [comment]
Do you have a good alternative? If we wouldn't have the volume in the Dockerfile by default, people would lose their database / files every time the container is recreated. That's not desirable either

**Comment by AKockWorth** [comment]
You're right, It is a rather impactful breaking change. Because the directory isn't an anonymous volume by default. Although I suspect most production workloads will make the volumes either named or persisted explicitly, that isn't a strong enough argument (since it is speculation) to potentially break existing users and losing data.

One alternative is to have the final image itself be derived from a parent container image (eg `directus/directus-parent`), where the `VOLUME` isn't set. Then users can choose to derive from that parent and customize from there.

Another alternative might be to have more granular control over the extensions path, so that, in addition to `EXTENSIONS_PATH` there is a `EXTENSIONS_HOOKS_PATH` etc.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
